lib/db/structs.go

```diff
@@ -129,19 +129,19 @@ func (f FileInfoTruncated) FileModifiedBy() protocol.ShortID {
 }
 
 func (f FileInfoTruncated) ConvertToIgnoredFileInfo(by protocol.ShortID) protocol.FileInfo {
-	file := f.copyToFileInfo()
+	file := f.CopyToFileInfo()
 	file.SetIgnored(by)
 	return file
 }
 
 func (f FileInfoTruncated) ConvertToDeletedFileInfo(by protocol.ShortID) protocol.FileInfo {
-	file := f.copyToFileInfo()
+	file := f.CopyToFileInfo()
 	file.SetDeleted(by)
 	return file
 }
 
 // copyToFileInfo just copies all members of FileInfoTruncated to protocol.FileInfo
-func (f FileInfoTruncated) copyToFileInfo() protocol.FileInfo {
+func (f FileInfoTruncated) CopyToFileInfo() protocol.FileInfo {
```

Comment:
Somewhat skeptical of exposing this method given the latest change to putFile and the fact that the resulting FileInfo can't be easily or safely passed back into the database layer. The Deleted/Ignored cases above are the exceptions where this is safe (as those shouldn't have block lists).

Reply:
I added a new method that converts deleted, truncated file infos to protocol file infos and panics if the file isn't actually deleted.